source/dotnet/Services/WebApi.UnitTests/Infrastructure/Integration/CalculationResultReadyIntegrationEventFactoryTests.cs

```diff
@@ -202,4 +202,93 @@ public void CreateCalculationResultCompletedForEnergySupplier_WhenQuantityQualit
             processCompletedEventDto,
             "AGlnNumber"));
     }
+
+    [Fact]
+    public void CreateCalculationResultCompletedForBalanceResponsibleParty_WhenCreating_ResultIsForBalanceResponsibleParty()
+    {
+        // Arrange
+        var sut = new CalculationResultReadyIntegrationEventFactory();
+        var processStepResultDto = new ProcessStepResult(
+            TimeSeriesType.Production,
+            new TimeSeriesPoint[] { new(DateTimeOffset.Now, 10.0m, QuantityQuality.Estimated) });
+        var processCompletedEventDto = new ProcessCompletedEventDto(
+            "805",
+            Guid.NewGuid(),
+            ProcessType.Aggregation,
+            Instant.FromUtc(2022, 5, 1, 0, 0),
+            Instant.FromUtc(2022, 5, 2, 0, 0));
+
+        // Act
+        var actual = sut.CreateCalculationResultCompletedForBalanceResponsibleParty(
+            processStepResultDto,
+            processCompletedEventDto,
+            "ABrpGlnNumber");
+
+        // Assert
+        actual.AggregationPerGridarea.Should().BeNull();
+        actual.AggregationPerEnergysupplierPerBalanceresponsiblepartyPerGridarea.Should().BeNull();
+        actual.AggregationPerEnergysupplierPerGridarea.Should().BeNull();
+        actual.AggregationPerBalanceresponsiblepartyPerGridarea.Should().NotBeNull();
+    }
+
+    [Fact]
+    public void CreateCalculationResultCompletedForBalanceResponsibleParty_WhenCreating_PropertiesAreMappedCorrectly()
+    {
+        // Arrange
+        var sut = new CalculationResultReadyIntegrationEventFactory();
+        var timeSeriesPoint = new TimeSeriesPoint(DateTimeOffset.Now, 10.101000000m, QuantityQuality.Estimated);
+        var processStepResultDto = new ProcessStepResult(
+            TimeSeriesType.Production,
+            new[] { timeSeriesPoint });
+        var processCompletedEventDto = new ProcessCompletedEventDto(
+            "805",
+            Guid.NewGuid(),
+            ProcessType.Aggregation,
+            Instant.FromUtc(2022, 5, 1, 0, 0),
+            Instant.FromUtc(2022, 5, 2, 0, 0));
+        var expectedGln = "TheBrpGln";
+
+        // Act
+        var actual = sut.CreateCalculationResultCompletedForBalanceResponsibleParty(
+            processStepResultDto,
+            processCompletedEventDto,
+            expectedGln);
+
+        // Assert
+        actual.AggregationPerBalanceresponsiblepartyPerGridarea.GridAreaCode.Should().Be(processCompletedEventDto.GridAreaCode);
```

Comment:
How about instantiating an "expected" in the arrange section and then:
```suggestion
        actual.Should().BeEquivalentTo(expected);
```